src/web/components/ApiKeyManagement/KeyCreationDialog.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState } from 'react';
+import { SubmitHandler } from 'react-hook-form';
+
+import { ApiRoleDTO } from '../../../api/entities/ApiRole';
+import { ApiKeySecretsDTO } from '../../../api/services/apiKeyService';
+import { CreateApiKeyFormDTO } from '../../services/apiKeyService';
+import { Dialog } from '../Core/Dialog';
+import { Form } from '../Core/Form';
+import KeySecretReveal from '../Core/KeySecretReveal';
+import { StatusPopup } from '../Core/StatusPopup';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { TextInput } from '../Input/TextInput';
+
+import './KeyCreationDialog.scss';
+
+type KeySecretProp = ApiKeySecretsDTO | undefined;
+
+type KeyCreationDialogProps = {
+  onKeyCreation: (form: CreateApiKeyFormDTO) => Promise<KeySecretProp>;
+  triggerButton: JSX.Element;
+  availableRoles: ApiRoleDTO[];
+};
+
+function KeyCreationDialog({
+  onKeyCreation,
+  triggerButton,
+  availableRoles,
+}: KeyCreationDialogProps) {
+  const [open, setOpen] = useState(false);
+  const [secrets, setSecrets] = useState<KeySecretProp>(undefined);
+  const [showStatusPopup, setShowStatusPopup] = useState<boolean>(false);
+  const [statusPopupMessage, setStatusPopupMessage] = useState<string>('');
+  const [copiedSecrets, setCopiedSecrets] = useState<Map<String, boolean>>(
+    new Map<String, boolean>()
+  );
+
+  const onFormSubmit: SubmitHandler<CreateApiKeyFormDTO> = async (formData) => {
+    setSecrets(await onKeyCreation(formData));
+    setStatusPopupMessage(
+      'Your key has been created. Please copy the credentials before closing the window.'
+    );
+    setShowStatusPopup(true);
+  };
+
+  function confirmClose(): void {
+    const values = [...copiedSecrets.values()];
+
+    if (values.filter((value) => value === false).length > 0) {
+      setStatusPopupMessage('Please copy all secrets shown before closing the page');
+      setShowStatusPopup(true);
+      return;
+    }
+
+    setSecrets(undefined);
+    setOpen(false);
+  }
+
+  return (
+    <div className='key-creation-dialog'>
+      <Dialog triggerButton={triggerButton} open={open} onOpenChange={setOpen} hideClose>
+        {!secrets ? (
```

Comment:
Would it make sense to extract some of this out rather than having a big ternary? Something like:
```
{!secrets ? <CreateApiKeyForm /> : <ApiSecrets />}
```
Just an idea to improve readability

Reply:
Have done this since saves having to define all the values. Happy to use actual components if you think that looks better.

```
{!secrets
          ? CreateApiKeyForm(onFormSubmit, availableRoles, setOpen)
          : ShowApiKeySecrets(
              secrets,
              setCopiedSecrets,
              confirmClose,
              showStatusPopup,
              setShowStatusPopup,
              statusPopupMessage
            )}
```

Reply:
I personally prefer actual components but perhaps not a big deal either way. Fine to leave as-is